AIPscan/Data/data.py

```diff
@@ -422,3 +437,44 @@ def aips_by_puid(storage_service_id, puid, original_files=True):
         original_files=original_files,
         file_format=False,
     )
+
+
+def agents_transfers(storage_service_id):
+    """Return information about agents based on event or events supplied
+    to the function.
+    """
+    report = {}
+    ingests = []
+
+    storage_service = _get_storage_service(storage_service_id)
+
+    try:
+        report[FIELD_STORAGE_NAME] = storage_service.name
+    except AttributeError:
+        # No storage service has been returned and so we have nothing
+        # to return.
+        report[FIELD_STORAGE_NAME] = None
+        report[FIELD_INGESTS] = ingests
+        return report
+
+    aips = AIP.query.filter_by(storage_service_id=storage_service.id).all()
+
+    EVENT_TYPE = "ingestion"
+    AGENT_TYPE = "Archivematica user"
+
+    for aip in aips:
+        files = File.query.filter_by(aip_id=aip.id)
+        for file in files[:1]:
+            events = Event.query.filter_by(file_id=file.id, type=EVENT_TYPE).all()
+            for event in events:
+                log_line = {}
+                log_line[FIELD_AIP] = aip.uuid
+                log_line[FIELD_AIP_NAME] = aip.transfer_name
+                log_line[FIELD_INGEST_START_DATE] = str(event.date)
+                log_line[FIELD_INGEST_FINISH_DATE] = str(aip.create_date)
```

Comment:
This is interesting. It makes perfect sense as implemented. But it makes me wonder about which of these values we should be using for the start and end date parameters on our reports, and how we communicate that choice.

For this particular report, I imagine we'd want to use the ingestion `Event` date for start date, and `AIP.create_date` for end date, to keep it consistent.

For other reports, I've been using `AIP.create_date` for both dates in my WIP code (rationale being that it reflects what's in the Storage Service/Archival Storage tab) but it looks like we've used the ingestion `Event` date for both elsewhere and I can understand why that's also a reasonable choice. Granted, in most cases they'll be the same since those parameters only accept YYYY-MM-DD values, but they might not be for some very long-running ingests. What do you think?

Reply:
I wonder if you can clarify that comment/question a little Tessa? Primarily: 

> For this particular report, I imagine we'd want to use the ingestion Event date for start date, and AIP.create_date for end date, to keep it consistent.

Which I think I am doing?

Reply:
Yes! To be clear I don't think anything in this PR needs to change.

I meant this more as a conversation prompt for AIPscan as a whole, looking ahead to adding `start_date` and `end_date` parameters for more reports in the reports UI/API, including potentially this one.

As I said above, I think how you've implemented the dates for this report makes perfect sense, and it would make sense to continue by applying the same logic to date inputs that come in with the request from the GUI or API. But that would mean that those parameters would be a little different to how they are implemented in the "File format countt" (where the ingestion `Event` date is used for both `start_date` and `end_dates`), which is also different to how I had been thinking about it in relation to the "Format version count" report I'm about to submit a PR for (where I've using `AIP.create_date`, as the date that the user would see in their Archival Storage tab/Storage Service and might want to carry over into AIPscan).

It's more a question than a prompt for change, and maybe we can hash it out in other PRs or conversations, but I thought that since your work prompted the thought and it potentially has an impact on the work I'm just about to submit, I'd check if you had any initial thoughts.

Reply:
With the same caveat as above that I don't intend these as a blocker for this PR, I guess another way to think about this might be - if a transfer is created in January and stored in the transfer backlog, and then that transfer (or parts of it, or parts of several transfers) get run through ingest and turned into an AIP in June, the ingestion event and the AIP creation date will be very different. When we say "start date" and "end date" in relation to something like the "File format count" report, do we mean when the file was first ingested into an Archivematica instance, or when it became part of an AIP? Might that answer change for other reports? Is there a way to communicate what it means to the user?

Reply:
@ross-spencer I'm marking this as resolved since it shouldn't get in your way, and we can have the conversation in the PR I'm just about to submit: https://github.com/artefactual-labs/AIPscan/pull/98

Reply:
Oh cool. Yeah, these are all good points. Initial instinct at a low level is to encode some of this in helpers in AIPscan at least that describe the heuristic and values being retrieved more completely than I've done here (that will also promote greater accuracy). On a higher-level Joel recognized early on that we might want a domain model for this piece. (Well, all parts of our work really). But this is a good early catch, thanks for putting in the forefront of my mind too.